pennylane/labs/trotter/realspace/realspace_operator.py

```diff
@@ -0,0 +1,407 @@
+
+        return RealspaceSum(self.modes, new_terms)
+
+    def __mul__(self, scalar: float) -> RealspaceSum:
+        return RealspaceSum(self.modes, [scalar * term for term in self.ops])
+
+    __rmul__ = __mul__
+
+    def __imul__(self, scalar: float) -> RealspaceSum:
+        for term in self.ops:
+            term *= scalar
+
+        return self
+
+    def __matmul__(self, other: RealspaceSum) -> RealspaceSum:
+        return RealspaceSum(
+            self.modes,
+            [
+                RealspaceOperator(
+                    self.modes, l_term.ops + r_term.ops, l_term.coeffs @ r_term.coeffs
+                )
+                for l_term, r_term in product(self.ops, other.ops)
+            ],
+        )
+
+    def __repr__(self) -> str:
+        return self.ops.__repr__()
+
+    def __eq__(self, other: RealspaceSum) -> bool:
+        return self._lookup == other._lookup
+
+    @classmethod
+    def zero(cls, modes: int) -> RealspaceSum:
+        """Returns a RealspaceOperator representing the zero operator
+
+        Args:
+            modes (int): the number of vibrational modes (needed for consistency with arithmetic operations)
+
+        Returns:
+            RealspaceOperator: a representation of the zero operator
+
+        """
+        return RealspaceSum(modes, [RealspaceOperator.zero(modes)])
+
+    def matrix(
+        self, gridpoints: int, basis: str = "realspace", sparse: bool = False
+    ) -> Union[np.ndarray, sp.sparse.cs_array]:
+        """Return a matrix representation of the ``RealspaceSum``.
+
+        Args:
+            gridpoints (int): the number of gridpoints used to discretize the position/momentum operators
+            basis (str): the basis of the matrix, available options are ``realspace`` and ``harmonic``
+            sparse (bool): if True returns a sparse matrix, otherwise a dense matrix
+
+        Returns:
+            Union[ndarray, csr_array]: the matrix representation of the ``RealspaceOperator``
+
+        """
+
+        final_matrix = _zeros(shape=(gridpoints**self.modes, gridpoints**self.modes), sparse=sparse)
+        for op in self.ops:
+            final_matrix = final_matrix + op.matrix(gridpoints, basis=basis, sparse=sparse)
+
+        return final_matrix
+
+    def norm(self, params: Dict) -> float:
+        """Returns an upper bound on the spectral norm of the operator.
+
+        Args:
+            params (Dict): The dictionary of parameters. The supported parameters are
+
+                * ``gridpoints`` (int): the number of gridpoints used to discretize the operator
+                * ``sparse`` (bool): If True, use optimizations for sparse operators. Defaults to False.
+
+        Returns:
+            float: an upper bound on the spectral norm of the operator
+
+        """
+
+        try:
+            gridpoints = params["gridpoints"]
+        except KeyError as e:
+            raise KeyError("Need to specify the number of gridpoints") from e
```

Comment:
```suggestion
        gridpoints = params.get("gridpoints", None)
        if gridpoints is None:
            raise KeyError("Need to specify the number of gridpoints")
```
This is cleaner and a bit more efficient.